test/Microsoft.DotNet.ToolPackage.Tests/ToolPackageObtainerTests.cs

```diff
@@ -207,6 +212,52 @@ public void GivenNonExistentNugetConfigFileItThrows()
 
         }
 
+        [Fact]
+        public void GivenSourceItPassesOptionToRestoreCommand()
+        {
+            string name = null;
+            IEnumerable<string> args = null;
+            const string expectedSource = "TestSource";
+
+            var mockCommand = new Mock<ICommand>();
+            mockCommand
```

Comment:
In this adapter layer we should not test with mock nuget(restore), and maybe this is the reason run out of proc worked but it will failed on production.

Adapter layer by definition coupled with external service(nuget, also nuget is in rapid development). Test with mock cannot gain enough confident..

Either you need to prove the mock behave the same as nuget(have the same test run again the mock and real nuget), or we should just run again real nuget, in CLI case, although the test are relatively slow, it is reliable.

Reply:
After our conversation, I'll address your concerns here by removing the mocking.